dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/models/SubSnpCoreFields.java

```diff
@@ -0,0 +1,151 @@
+
+    /**
+     * @param ssId Unique SS ID identifier
+     * @param rsId Unique RS ID identifier, can be null if the SS ID has not been clustered yet
+     * @param snpOrientation Orientation of the SS ID (forward/reverse)
+     * @param contig Contig name
+     * @param contigStart Start coordinate in contig
+     * @param contigEnd End coordinate in contig
+     * @param contigOrientation Orientation of the contig (forward/reverse)
+     * @param chromosome Chromosome name, can be null if the contig is not mapped to a chromosome
+     * @param chromosomeStart Start coordinate in chromosome, null if the contig is not fully mapped to a chromosome
+     * @param chromosomeEnd End coordinate in chromosome, null if the contig is not fully mapped to a chromosome
+     */
+    public SubSnpCoreFields(long ssId, Integer rsId, int snpOrientation, String contig, int contigStart, int contigEnd,
+                            int contigOrientation, String chromosome, Long chromosomeStart, BigDecimal chromosomeEnd) {
+        if (contigStart < 0 || contigEnd < 0) {
+            throw new IllegalArgumentException("Contig coordinates must be non-negative numbers");
+        }
+        if ((chromosomeStart != null && chromosomeStart < 0)
+                || (chromosomeEnd != null && chromosomeEnd.compareTo(new BigDecimal(0)) == 0)) {
+            throw new IllegalArgumentException("Chromosome coordinates must be non-negative numbers");
+        }
+
+        this.ssId = ssId;
+        this.rsId = rsId;
+        this.contigRegion = createRegion(contig, (long) contigStart, BigDecimal.valueOf(contigEnd));
+        this.chromosomeRegion = createRegion(chromosome, chromosomeStart, chromosomeEnd);
+        this.snpOrientation = Orientation.getOrientation(snpOrientation);
+        this.contigOrientation = Orientation.getOrientation(contigOrientation);
+    }
+
+    public long getSsId() {
+        return ssId;
+    }
+
+    public Integer getRsId() {
+        return rsId;
+    }
+
+    public Region getContigRegion() {
+        return contigRegion;
+    }
+
+    public Region getChromosomeRegion() {
+        return chromosomeRegion;
+    }
+
+    public Orientation getSnpOrientation() {
+        return snpOrientation;
+    }
+
+    public Orientation getContigOrientation() {
+        return contigOrientation;
+    }
+
+    private Region createRegion(String sequenceName, Long start, BigDecimal end) {
+        if (sequenceName == null) {
+            // This should happen only with chromosomes, when a contig-to-chromosome mapping is not available
+            return null;
+        }
+
+        StringBuilder regionString = new StringBuilder(sequenceName);
+        if (start != null) {
+            regionString.append(":");
+            regionString.append(start);
+
+            if (end != null) {
+                regionString.append(":");
+                regionString.append(end);
+            }
+        }
+
+        return new Region(regionString.toString());
```

Comment:
I guess we use this constructor because Region class just admit integers and not long values

Reply:
That's it. I think we should modify the `Region` class to make it support `long` values, to avoid losing any data during the import even when we are reading it correctly :confused:

Reply:
we won't lose data in the import even with ints, although I would prefer having coordinates as longs in the medium-long term